packages/backend/src/core/entities/NotificationEntityService.ts

```diff
@@ -140,7 +139,10 @@ export class NotificationEntityService implements OnModuleInit {
 		// #endregion
 
 		const needsRole = notification.type === 'roleAssigned';
-		const role = needsRole ? await this.roleEntityService.pack(notification.roleId) : undefined;
+		const role = needsRole ? await this.roleEntityService.pack(notification.roleId).catch((e) => {
+			if (e.name === 'EntityNotFoundError') return null;
+			throw e;
```

Comment:
```suggestion
		const role = needsRole ? await this.roleEntityService.pack(notification.roleId).catch((err) => {
			if (err.name === 'EntityNotFoundError') return null;
			throw err;
```

`e`という名前は混同が起こりやすいので禁止されていた気がします

Reply:
あと、ここは文字列比較にするより下記のように `instanceof` を使ったほうが保守性が良くなると思います。
https://github.com/misskey-dev/misskey/blob/74c93fcebe1b6fde489470e19808389d13f07a05/packages/backend/src/server/api/endpoints/announcements/show.ts#L49

Reply:
レビューありがとうございます。
instanceofとe両方修正入れました。

例ではApiErrorを利用していますが、ここはApiではないのと、本来この直後はロールが削除されていたときにnullが返ってくることが前提の作りになっていたためその意向に従い、return nullでroleにnullを突っ込んで処理させた方が良いと思い、return nullのままにしております。